# 🧬 Mycoplasma-assembly-pipeline
This repository contains a 𝐰𝐡𝐨𝐥𝐞 𝐠𝐞𝐧𝐨𝐦𝐞 𝐚𝐬𝐬𝐞𝐦𝐛𝐥𝐲 𝐩𝐢𝐩𝐞𝐥𝐢𝐧𝐞 implemented on Linux for analyzing **𝘔𝘺𝘤𝘰𝘱𝘭𝘢𝘴𝘮𝘢 𝘱𝘯𝘦𝘶𝘮𝘰𝘯𝘪𝘢𝘦** sequencing data using Illumina MiSeq reads.

---

## Table of contents
<ul>
  <li><a href="#overview">Overview</a></li>
  <li><a href="#dataset">Dataset</a></li>
  <li><a href="#tools--technologies">Tools & Technologies</a></li>
  <li><a href="#project-structure">Project Structure</a></li>
  <li><a href="#pipeline-workflow">Pipeline Workflow</a></li>
  <li><a href="#before--after-filtering-results">Before & After Filtering Results</a></li>
  <li><a href="#results-summary">Results Summary</a></li>
  <li><a href="#bandage-graph-visualization">Bandage Graph Visualization</a></li>
  <li><a href="#key-concepts">Key Concepts</a></li>
  <li><a href="#how-to-run-this-project">How to Run This Project</a></li>
  <li><a href="#future-work">Future Work</a></li>
  <li><a href="#author--contact">Author & Contact</a></li>
  <li><a href="#license">License</a></li>
</ul>

---
## Overview
This project demonstrates a complete whole genome assembly workflow for **_Mycoplasma pneumoniae_**, starting from raw Illumina sequencing data to final assembly evaluation. The pipeline integrates standard open-source bioinformatics tools to ensure high-quality assemblies and reproducible results.

---

## Dataset

**Organism:** Mycoplasmoides pneumoniae (strain FH_USA_MA_2009)

**Sequencing platform:** Illumina MiSeq

**Library type:** Long-insert shotgun, paired-end

**Input Data:** SRR3924617 (~1.36M spots, ~393 Mb bases, 183 Mb download size)

**Output:** Genome assembly (FASTA), visualization (Bandage), and quality reports (QUAST)

---

## Tools & Technologies

**SRA Toolkit**: Convert .sra to FASTQ

**FastQC**: Raw read quality check

**Fastp**: Adapter trimming & quality filtering

**SPAdes**: Genome assembly

**QUAST**: Assembly quality evaluation

**Bandage**: Assembly graph visualization

**Bash**: Pipeline automation

---
## Project Structure
```bash
mycoplasma-assembly-pipeline/
│
├── Data/            # Raw sequencing data (.sra, FASTQ)
├── Quality/         # FastQC results
├── Trimmed/         # fastp results
├── Assembly/        # SPAdes output (contigs, scaffolds, log files)
├── QUAST/           # QUAST report and assembly statistics
├── Bandage/         # Bandage graph visualization (assembly graph)
├── Documents/       # Flowchart & figures
└── Script/          # Command pipeline 
```

<p align="center">
  <img src="https://github.com/genome-miner/mycoplasma-assembly-pipeline/blob/main/Documents/Flowchart.png?raw=true" alt="Alt text" width="150">
</p>

---

## Before & After Filtering Results:


## Results Summary
Assembly statistics (QUAST report, contigs ≥500 bp):

| Metric            | Value      |
| ----------------- | ---------- |
| Contigs (≥500 bp) | 14         |
| Contigs (total)   | 15         |
| Largest contig    | 324,076 bp |
| Total length      | 799,172 bp |
| GC content        | 39.78%     |
| N50               | 92,062 bp  |
| N75               | 62,710 bp  |
| L50               | 2          |
| L75               | 5          |
| N’s per 100 kbp   | 0.00       |

---

## Bandage Graph Visualization
The assembly graph (after simplification) generated by SPAdes was visualized using Bandage:
**Nodes:** 551
**Edges:** 684
**Total graph length:** 857,023 bp
This graph provides a visual representation of genome assembly connectivity, identifying possible repeats, unresolved regions, and structural complexities.

---

## Key Concepts
1. Genome Assembly
2. Illumina Sequencing
3. SPAdes & QUAST Analysis
4. Assembly Graph Interpretation
5. Bioinformatics Pipeline Automation

## How to Run This Project
**Clone this repository:**
```bash
git clone https://github.com/genome-miner/mycoplasma-assembly-pipeline.git
cd mycoplasma-assembly-pipeline
```
Download sequencing data using the SRA Toolkit.

Run pipeline scripts inside the [Script](https://github.com/genome-miner/mycoplasma-assembly-pipeline/tree/main/Script) folder.

Explore outputs in [Assembly](https://github.com/genome-miner/mycoplasma-assembly-pipeline/tree/main/Assembly), [QUAST](https://github.com/genome-miner/mycoplasma-assembly-pipeline/tree/main/Quast), and [Bandage](https://github.com/genome-miner/mycoplasma-assembly-pipeline/tree/main/Bandage).

---

## Future Work
While this pipeline successfully assembles and evaluates the Mycoplasma pneumoniae genome, several improvements and extensions are possible:
1. Genome annotation: Predict genes, proteins, and functional elements using tools such as Prokka.
2. Comparative genomics: Align assembled genome to reference strains to identify mutations, insertions, or deletions.
3. Phylogenomic analysis: Place assembled genome in evolutionary context with related species.
4. Extension to other organisms: Adapt the pipeline for bacterial or viral genome assembly beyond Mycoplasma.

---

## Author & Contact
Sana Aziz Sial  
Biotechnologist and Bioinformatician  
📧 Email: biotechnologist09pharma@gmail.com  
🔗 [LinkedIn](in/sana-aziz-sial-73b189265)  

_Open to collaboration and feedback!_  

---

## License
This project is licensed under the [MIT License](https://github.com/genome-miner/mycoplasma-assembly-pipeline/blob/main/LICENSE).
